src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -201,8 +219,41 @@ public void trimOldSnapshots() throws Exception {
     trimMap(shardRqMetricsMap, RQ_SNAPSHOTS);
     trimMap(httpRqMetricsMap, HTTP_RQ_SNAPSHOTS);
     trimMap(masterEventMetricsMap, MASTER_EVENT_SNAPSHOTS);
-    trimDatabases(
-        metricsDBMap, MAX_DATABASES, PluginSettings.instance().shouldCleanupMetricsDBFiles());
+
+    boolean deleteDBFiles = PluginSettings.instance().shouldCleanupMetricsDBFiles();
+    if (metricsDBMap.size() > MAX_DATABASES) {
+      Map.Entry<Long, MetricsDB> lowestEntry = metricsDBMap.pollFirstEntry();
+      if (lowestEntry != null) {
+        Long key = lowestEntry.getKey();
+        MetricsDB value = lowestEntry.getValue();
+        value.remove();
+        if (deleteDBFiles && !batchMetricsDBSet.contains(key)) {
+          value.deleteOnDiskFile();
```

Comment:
This logic should be placed into the trimDatabases function

Reply:
trimDatabases currently doesn't accept a parameter that would allow us to retain the db if it is in batchMetricsDBSet. trimDatabases could be modified to include such logic. However, such a change wouldn't really contribute much, as trimDatabases is only used in timOldSnapshots. Moving it out of trimOldSnapshots would also not even help with testing, as there are two other different trimming-related procedures in trimOldSnapshots.